static/app/views/dashboardsV2/detail.tsx

```diff
@@ -318,11 +318,18 @@ class DashboardDetail extends Component<Props, State> {
 
   /**
    * Saves a dashboard layout where the layout keys are replaced with the IDs of new widgets.
+   *
+   * If there are more widgets than layout objects, these widgets will be treated as
+   * new and will get default positioning. This happens when saving in mobile
+   * view because we don't update the desktop layout to account for the new widget.
+   *
+   * Throws an error if we end up in a state where we're trying to save more layouts
+   * than we have widgets.
    */
   saveLayoutWithNewWidgets = (organizationId, dashboardId, newWidgets) => {
     const {layout} = this.state;
-    if (layout.length !== newWidgets.length) {
-      throw new Error('Expected layouts and widgets to be the same length');
+    if (layout.length > newWidgets.length) {
```

Comment:
This might throw and error when you delete a widget in mobile view then?

Reply:
Oh actually, tested it out. It also results in a weird state where when you delete the widget and hit save. It throws the uncaught error and keeps you in edit view, but it looks like the dashboard has been updated so cancel doesn't roll back the changes.

Reply:
Ah, that's not supposed to happen. I originally made this check because to translate layouts from using grid keys with `tempId`s to using real `id`s, I made some assumptions that the widgets were 1:1 with the layouts

I fixed this by removing the positioning for `widgetToDelete` from the layout we intend to save
https://github.com/getsentry/sentry/pull/30595/files#diff-e36bb7a10b73d567eb9c53d02080462f63f87071a4f2c5a5e5829dbe99b11054R212